### PR TITLE
Align credentials stores- part 3

### PR DESCRIPTION
**Pull request by maimorag**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
Relates:[ link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-5274)

## Description
Change the type of credentials, from type 4 to type 9.

## Minimum version of Cortex XSOAR
- [x] 6.5.0

## Does it break backward compatibility?
 - [x] No

### CODE REVIEW

File: Packs/BeyondTrust_Password_Safe/Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 USERNAME = demisto.params().get('credentials', {}).get('identifier')
 PASSWORD = demisto.params().get('credentials', {}).get('password')
-API_KEY = demisto.params().get('key')
+API_KEY = demisto.params().get('key') or demisto.params().get('credentials_key', {}).get('password')
```
Comment (Shellyber) [inline Packs/BeyondTrust_Password_Safe/Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py:19]:
```suggestion
API_KEY = demisto.params().get('credentials_key', {}).get('password') or demisto.params().get('key')
```

**Review by Shellyber — COMMENTED** [review]
The new addition should be first - see my comment, change everywhere it is needed

File: Packs/BeyondTrust_Password_Safe/Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 2]
```diff
@@ -16,7 +16,9 @@
 
 USERNAME = demisto.params().get('credentials', {}).get('identifier')
 PASSWORD = demisto.params().get('credentials', {}).get('password')
-API_KEY = demisto.params().get('key')
+API_KEY = demisto.params().get('credentials_key', {}).get('password') or demisto.params().get('key')
+if not API_KEY or not USERNAME or not PASSWORD:
```
Comment (Shellyber) [inline Packs/BeyondTrust_Password_Safe/Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py:20]:
```suggestion
if not API_KEY or not (USERNAME and PASSWORD):
```

File: Packs/BeyondTrust_Password_Safe/Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 3]
```diff
@@ -16,7 +16,9 @@
...
+    raise DemistoException('API Key must be required')
```
Comment (Shellyber) [inline Packs/BeyondTrust_Password_Safe/Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py:21]:
```suggestion
    raise DemistoException('The API Key or username and password must be provided')
```

File: Packs/BeyondTrust_Password_Safe/Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml [hunk 4]
```diff
@@ -9,12 +9,18 @@ configuration:
   type: 0
 - display: Username
   name: credentials
-  required: true
+  required: false
```
Comment (Shellyber) [inline Packs/BeyondTrust_Password_Safe/Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml:12]: Why did you change this to False?

File: Packs/BeyondTrust_Password_Safe/Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py (hunk 2 above)
Comment (Shellyber) [reply Packs/BeyondTrust_Password_Safe/Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py:20]: OR it requires all three to authenticate?

File: Packs/MailListener_-_POP3/Integrations/MailListener_POP3/MailListener_POP3.py [hunk 5]
```diff
@@ -13,7 +13,9 @@
 ''' GLOBALS/PARAMS '''
 SERVER = demisto.params().get('server', '')
 EMAIL = demisto.params().get('email', '')
-PASSWORD = demisto.params().get('password', '')
+PASSWORD = demisto.params().get('credentials_password', {}).get('password') or demisto.params().get('password')
+if not PASSWORD:
+    raise DemistoException('Password must be required')
```
Comment (Shellyber) [inline Packs/MailListener_-_POP3/Integrations/MailListener_POP3/MailListener_POP3.py:18]:
```suggestion
    raise DemistoException('Password must be provided')
```

File: Packs/BeyondTrust_Password_Safe/Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml (hunk 4 above)
Comment (maimorag) [reply Packs/BeyondTrust_Password_Safe/Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml]: mistake

**Review by Shellyber — APPROVED** [review]
